storage/interface.go

```diff
@@ -271,6 +275,9 @@ type Appender interface {
 	// Appender has to be discarded after rollback.
 	Rollback() error
 
+	// New SetHints method to set the append hints.
```

Comment:
Please don't use "New" in the code: things stop being new as soon as you commit them.

```suggestion
	// SetHints method to set the append hints.
```

Also: I think this method deserves a better comment: why would you want to set hints? What do they affect? Can I call with `nil`? What if I call several times?